packages/dataclass-editor/src/DataClassEditor.tsx

```diff
@@ -41,7 +56,40 @@ function DataClassEditor(props: EditorProps) {
     return <PanelMessage icon={IvyIcons.ErrorXMark} message={`An error has occurred: ${error.message}`} />;
   }
 
-  return <DataClassMaster dataClassFields={data.data.fields} />;
+  const dataClass = data.data;
+  const dataClassFields = dataClass.fields;
+
+  const title = isEntityClass(dataClass) ? 'Entity Class Editor' : 'Data Class Editor';
+  let detailTitle = title;
+  if (selectedDataClassFieldIndex !== undefined && selectedDataClassFieldIndex < dataClassFields.length) {
+    const selectedDataClassField = dataClassFields[selectedDataClassFieldIndex];
+    detailTitle = 'Attribute - ' + selectedDataClassField.name;
+  }
+
+  return (
+    <ResizablePanelGroup direction='horizontal' style={{ height: `100vh` }}>
+      <ResizablePanel defaultSize={75} minSize={50} className='master-panel' data-testid='master-panel'>
+        <Flex className='panel-content-container' direction='column'>
+          <Toolbar className='master-toolbar'>
+            <ToolbarTitle>{title}</ToolbarTitle>
+            <DataClassMasterToolbarControls sidebar={sidebar} setSidebar={setSidebar} />
+          </Toolbar>
```

Comment:
I would move also the title into the toolbar component and just call it `DataClassToolbar`. Because it is the responsibility of this toolbar to 'calulate' the title.
Currently you use this title also for the sidebar if no element is selected, but then I would simply name the title 'Attribute' without the '- name'.

Reply:
Unlike the Variables Editor, if no element is selected, you can edit the properties of the data class itself in the detail view. According to the [Figma mock-up](https://www.figma.com/design/kJpTtg11jKnwYknbkgiuwD/Full-Stack-Designer?node-id=608-252929&t=wh2jsoo1LeFfDQp6-0), the detail title should be `Data Class Editor` in this case (although I think `Data Class` would fit better).

This means the calculation of the master and detail titles have some overlapping logic, particularly, checking whether the data class is an entity class. However, I think it would be fine to duplicate this cheap and simple statement in a new component `DataClassToolbar`.

I will implement some react contexts with the detail view implementation and see how it works out.